x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/job/persistence/CalendarQueryBuilder.java

```diff
@@ -64,6 +65,7 @@ public SearchSourceBuilder build() {
         if (jobId != null) {
             jobIdAndGroups.add(jobId);
         }
+        jobIdAndGroups.add(MetaData.ALL);
```

Comment:
this should be moved under the `if (jobIdAndGroups.isEmpty() == false) {` clause.